### PR TITLE
Add UI for Patch Set Descriptions

### PR DESCRIPTION
**Pull request by monitorlizard**
Hi Urs

I love your plugin and use it daily. Here is the implementation of the issue I created I while ago:
https://github.com/uwolfer/gerrit-intellij-plugin/issues/426

Hope you like it!
Stephan

### CODE REVIEW

File: src/main/java/com/urswolfer/intellij/plugin/gerrit/util/UrlUtils.java [hunk 1]
```diff
@@ -43,4 +45,11 @@ public static URI createUriFromGitConfigString(String gitConfigUrl) {
     public static String stripGitExtension(String url) {
         return url.replace(".git", ""); // some repositories end their name with ".git"
     }
+
+    public static String encodePatchSetDescription(String text) {
+        return URLEncoder.encode(text, StandardCharsets.UTF_8)
+            .replaceAll("\\.", "%2E")
+            .replaceAll("-", "%96")
```
Comment (uwolfer) [inline src/main/java/com/urswolfer/intellij/plugin/gerrit/util/UrlUtils.java]: Have you found any documentation which chars need to be replaced? I would move the comment from the test up here, and add a link if possible.

Also note that `replaceAll` requires a regex - I do not think you are currently using a regex here?

**Review by uwolfer — COMMENTED** [review]
Thanks for this PR! I have not tested it yet - just added comments so far.

File: src/main/java/com/urswolfer/intellij/plugin/gerrit/util/UrlUtils.java (hunk 1 above)
Comment (monitorlizard) [reply src/main/java/com/urswolfer/intellij/plugin/gerrit/util/UrlUtils.java]: True, I was thinking about moving the comment as well. 
I will add the link to the comment: https://gerrit-review.googlesource.com/Documentation/user-upload.html#patch_set_description

Yes, you are right, it's not necessary to use RegEx, I'll change it to replace(...)

**Comment by uwolfer** [comment]
@monitorlizard Thank you for your work!

File: src/main/java/com/urswolfer/intellij/plugin/gerrit/push/GerritPushExtensionPanel.java [hunk 2]
```diff
@@ -200,6 +202,11 @@ private void createLayout() {
                 "Include a hashtag associated with all of the changes in the same group.",
                 9);
 
+        patchsetDescriptionTextField = addTextField(
+                "Patch Set Description (Gerrit 3.4+):",
+                "A description of the patch set to be created. Intended to help guide reviewers as a change evolves. The description cannot be changed after the change is pushed.",
+                10);
```
Comment (uwolfer) [inline src/main/java/com/urswolfer/intellij/plugin/gerrit/push/GerritPushExtensionPanel.java:208]: I have missed this number overlap with the next one - it causes an UI glitch. Would you like to create a PR?

See also: https://github.com/uwolfer/gerrit-intellij-plugin/issues/502

Comment (uwolfer) [reply src/main/java/com/urswolfer/intellij/plugin/gerrit/push/GerritPushExtensionPanel.java:208]: I have fixed it.

**Pull request merged**